spec/models/service_orchestration/provision_tagging_spec.rb

```diff
@@ -0,0 +1,83 @@
+describe ServiceOrchestration::ProvisionTagging do
+  shared_examples_for 'service_orchestration VM tagging' do
+    it 'assign tags' do
+      expect(miq_request_task).to receive(:provision_priority).and_return(provision_priority)
+      expect(Classification).to receive(:bulk_reassignment).with(
+        :model      => 'Vm',
+        :object_ids => [vm.id],
+        :add_ids    => tag_ids
+      )
+
+      service.post_provision_configure
+    end
+  end
+
+  describe '#apply_provisioning_tags' do
+    before { expect(service).to receive(:assign_vms_owner) }
+
+    let(:miq_request_task) { FactoryGirl.create(:service_template_provision_task) }
+    let(:vm) { FactoryGirl.create(:vm) }
+    let(:service) { FactoryGirl.build(:service_orchestration, :miq_request_task => miq_request_task) }
+    let(:dialog_tag_options) do
+      { :dialog => {
+        'Array::dialog_tag_0_env'     => 'Classification::1',
+        'Array::dialog_tag_1_network' => 'Classification::11',
+        'Array::dialog_tag_2_dept'    => 'Classification::21,Classification::22,Classification::23'
+      }}
+    end
+
+    context 'without service dialog tag options' do
+      it 'does not apply tags' do
+        expect(Classification).to receive(:bulk_reassignment).never
+
+        service.post_provision_configure
+      end
+    end
+
+    context `with a vm` do
```

Comment:
@gmcculloug Should these be apostrophes or a back tick.

Reply:
Good catch, my markdown past haunts me.  I'll clean up both, thanks.